src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1547,5 +1622,45 @@ internal override SyntaxNode ScopeDesignator
                 return _syntax;
             }
         }
+
+        private MethodArgumentInfo GetSpecialTypeMemberInfo(SpecialMember member, SyntaxNode syntax, DiagnosticBag diagnostics)
```

Comment:
>GetSpecialTypeMemberInfo [](start = 35, length = 24)

From this name it is not obvious that it can handle only methods without parameters.

Reply:
I'll also add an assert that the result was null or parameterless.

---
In reply to: [549837359](https://github.com/dotnet/roslyn/pull/50105#discussion_r549837359) [](ancestors = 549837359)